### PR TITLE
Only reordering and grouping of methods in the interface, object and file.

### PR DESCRIPTION
**Pull request by sea1jxr**
I am sure I didn't get this 100% correct, so fire away with your suggestions for improvements :)

### CODE REVIEW

**Comment by Jason3S** [comment]
Some of them are not easy to sort into categories.

These are really about extending the length of a sequence: (I'm not sure what to call them)
* `concat` extends a sequence by another sequence (sort of opposite of a filter).
* `concatMap` is more of a flattener. The resulting sequence can be longer than the original sequence.

**Comment by sea1jxr** [comment]
Yeah, I struggled with concat and concatMap for the far too long, so I finally just put them "anywhere" and was hoping you had a good category for them.

I kind of think of concat as a SET operation (much like union), but I can't really find any evidence that fits in that category.

File: src/GenSequence.ts [hunk 1]
```diff
@@ -7,22 +7,23 @@ export interface IterableLike<T> {
...
-    /** reduce function see Array.reduce */
-    reduce(fnReduce: (previousValue: T, currentValue: T, currentIndex: number) => T): Maybe<T>;
-    reduce<U>(fnReduce: (previousValue: U, currentValue: T, currentIndex: number) => U, initialValue: U): U;
-    reduceToSequence<U, V extends GenIterable<U>>(fnReduce: (previousValue: V, currentValue: T, currentIndex: number) => V, initialValue: V): Sequence<U>;
-    reduceToSequence<U>(fnReduce: (previousValue: GenIterable<U>, currentValue: T, currentIndex: number) => GenIterable<U>, initialValue: GenIterable<U>): Sequence<U>;
-    scan(fnReduce: (previousValue: T, currentValue: T, currentIndex: number, array: T[]) => T, initialValue?: T): Sequence<T>;
-    scan<U>(fnReduce: (previousValue: U, currentValue: T, currentIndex: number, array: T[]) => U, initialValue: U): Sequence<U>;
-    combine<U, V>(fn: (t: T, u?: U) => V, j: Iterable<U>): Sequence<V>;
     concat(j: Iterable<T>): Sequence<T>;
     concatMap<U>(fn: (t: T) => Iterable<U>): Sequence<U>;
```
Comment (Jason3S) [inline src/GenSequence.ts]: Let's put `concat` and `concatMap` into their own category: **Extenders**

**Comment by sea1jxr** [comment]
I noticed that reduceToSequence wasn't in the readme, so I added it.  Please take a close look at the wording there and see if you want any changes.

**Comment by Jason3S** [comment]
Good catch with reduceToSequence

**Pull request merged**